src/bitwise/core.cljs

```diff
@@ -120,27 +116,21 @@
 (defn tick [state dt]
   state)
 
-(defn handle-animation-frame [time]
-  (.requestAnimationFrame js/window handle-animation-frame)
-  (reset! last-update time))
-
-(defn handle-game-loop [dt-chan key ref prev-time curr-time]
-  (let [elapsed (ms->sec (- curr-time prev-time))]
-    (go-loop [acc (+ @accumulator elapsed)
-              i 0]
-      (if (>= acc timestep)
-        (do
-          (>! dt-chan timestep)
-          (swap! game-state #'tick timestep)
-          (recur (- acc timestep) (inc i)))
-        (reset! accumulator acc)))))
+(defn handle-animation-frame [timestamp-chan time]
+  (.requestAnimationFrame js/window (partial handle-animation-frame timestamp-chan))
+  (async/offer! timestamp-chan time))
 
 (defn init [time]
-  (let [dt-chan (chan (async/sliding-buffer 1))]
+  (let [dt-chan (chan (async/sliding-buffer 1))
+        timestamp-chan (chan (async/sliding-buffer 1))]
     (set! dt-mult (async/mult dt-chan))
-    (reset! last-update time)
-    (add-watch last-update :dt (partial #'handle-game-loop dt-chan)))
-  (.requestAnimationFrame js/window handle-animation-frame)
+    (go-loop [prev-time (<! timestamp-chan)
```

Comment:
The downside to moving the go-loop inside of init is that init is only called once, so if you make any changes here, you'll have to reload the page entirely to consume it. I think this should be pretty stable once we getting working how we want it to, so that's probably an acceptable tradeoff.